site/content/extend/plugins/webapp/actions.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Redux Actions
+date: 2018-07-10T00:00:00-05:00
+subsection: Web App Plugins
+weight: 11
+---
+
+# Redux Actions
+
+When building web app plugins, it is common to want to perform actions that the web and mobile apps already support. The majority of these actions exist in [mattermost-redux](https://github.com/mattermost/mattermost-redux), our library of shared code between our different JavaScript clients.
+
+If you want to learn more about these actions, see the [contributor documentation on them](/contribute/redux/actions).
+
+Here we'll show how to use these actions with a plugin.
+
+## Prerequisites
+
+This guide assumes you have already set up your plugin development environment for web app plugins to match [mattermost-plugin-sample](https://github.com/mattermost/mattermost-plugin-sample). If not, follow the README instructions of that repository first, or [see the Hello, World! guide](/extend/plugins/webapp/hello-world).
+
+## Import mattermost-redux
+
+First, you'll need to add `mattermost-redux` as a dependency of your web app plugin.
+
+```bash
+cd /path/to/plugin/webapp
+npm install mattermost-redux
+```
+
+That will add `mattermost-redux` as a dependency in your `package.json` file, allowing it to be imported into any of your plugin's JavaScript files.
+
+## Use an Action
+
+Actions are used as part of components. To give components access to these actions, we pass them in as React props from the component's container `index.js` file. To demonstrate this, we'll create a new component.
+
+In the `webapp` directory, let's create a component folder called `action_example` and switch into it.
+
+```bash
+mkdir -p src/components/action_example
+cd src/components/action_example
+```
+
+In there, create two files: `index.js` and `action_example.jsx`. If you're not familiar with why we're creating these directories and files, [read the contributor documentation on using React with Redux](/contribute/redux/react-redux).
+
+Open up `action_example.jsx` and add the following:
+
+```javascript
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class ActionExample extends React.PureComponent {
+    static propTypes = {
+        user: PropTypes.object.isRequired,
+        patchUser: PropTypes.func.isRequired, // here we define the action as a prop
+    }
+
+    updateFirstName = () => {
+        const patchedUser = {
+            id: this.props.user.id,
+            first_name: 'Jim',
+        };
+
+        this.props.patchUser(patchedUser); // here we use the action
+    }
+
+    render() {
+        return (
+            <div>
+                {'The user\'s first name is ' + this.props.user.first_name}
+                <a
+                    href='#'
+                    onClick={this.updateFirstName}
+                >
+                    Click me to update the user's first name!
```

Comment:
It doesn't like the single quote here, it seems

![image](https://user-images.githubusercontent.com/13119842/44371254-07fab080-a4ac-11e8-8376-5331d81bb441.png)